multiple definition of TOWER_TOP. Fix with extern declaration in header
diff -ruN starcode-1.4.orig/src/trie.c starcode-1.4/src/trie.c
--- starcode-1.4.orig/src/trie.c	2020-11-02 03:59:09.000000000 -0800
+++ starcode-1.4/src/trie.c	2022-06-28 14:22:19.168067000 -0700
@@ -29,6 +29,8 @@
 #define PAD 5              // Position of padding nodes.
 #define EOS -1             // End Of String, for 'dash()'.
 
+gstack_t * const TOWER_TOP;
+
 // Translation table to insert nodes in the trie.
 //          ' ': PAD (5)
 //     'a', 'A': 1
diff -ruN starcode-1.4.orig/src/trie.h starcode-1.4/src/trie.h
--- starcode-1.4.orig/src/trie.h	2020-11-02 03:59:09.000000000 -0800
+++ starcode-1.4/src/trie.h	2022-06-28 14:21:07.925395000 -0700
@@ -54,7 +54,7 @@
 #define MAXBRCDLEN 1023     // Maximum barcode length.
 #define GSTACK_INIT_SIZE 16 // Initial slots of 'gstack'.
 
-gstack_t * const TOWER_TOP;
+extern gstack_t * const TOWER_TOP;
 
 int         check_trie_error_and_reset (void);
 int         count_nodes (trie_t*);
